src/phoenix/server/api/input_types/ChatCompletionInput.py

```diff
@@ -2,19 +2,36 @@
 
 import strawberry
 from strawberry import UNSET
+from strawberry.relay.types import GlobalID
 from strawberry.scalars import JSON
 
+from phoenix.server.api.types.TemplateLanguage import TemplateLanguage
+
 from .ChatCompletionMessageInput import ChatCompletionMessageInput
 from .GenerativeModelInput import GenerativeModelInput
 from .InvocationParameters import InvocationParameterInput
 from .TemplateOptions import TemplateOptions
 
 
-@strawberry.input
-class ChatCompletionInput:
+@strawberry.type
+class BaseChatCompletionInput:
```

Comment:
Is there something we concrete gain by splitting this up into a base input and inherited input classes other than a small amount of de-duplication? My preference is to undo this if the only benefit is de-duplication

Reply:
It's here to satisfy the type checker since `BaseChatCompletion` is used as an input type. An alternative would be to use protocols or union inputs.

Reply:
changed to union